toolchain/sem_ir/stringify_type.cpp

```diff
@@ -24,78 +24,60 @@ static auto GetTypePrecedence(InstKind kind) -> int {
   return 0;
 }
 
-auto StringifyTypeExpr(const SemIR::File& outer_sem_ir, InstId outer_inst_id)
+auto StringifyTypeExpr(const SemIR::File& sem_ir, InstId outer_inst_id)
     -> std::string {
   std::string str;
   llvm::raw_string_ostream out(str);
 
-  struct Step {
-    // The instruction's file.
-    const File& sem_ir;
+  class StepStack {
```

Comment:
Would it make sense to move this out of the function? That might enable more refactoring, such as splitting this function into smaller functions, at a later stage.

Reply:
Possibly, though that seems out of scope for this PR.

Reply:
Oh I hadn't realized how little else was going on in this file. I've moved it out into an anonymous namespace.